src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -51,6 +59,12 @@ public TElement[] ToArray()
                 array[i] = buffer._items[map[i]];
             }
 
+            var rentedMap = Interlocked.Exchange(ref map, null);
```

Comment:
Unlike in the iterator, where `map` is really a field and different threads could call `MoveNext()`, `map` is a true local here. There's no risk of another thread hitting this, so no need for `Interlocked`.